crates/miden-testing/tests/auth/multisig.rs

```diff
@@ -0,0 +1,414 @@
+fn setup_keys_and_authenticators(
+    num_approvers: usize,
+    threshold: usize,
+) -> anyhow::Result<MultisigTestSetup> {
+    let mut rng = ChaCha20Rng::from_seed([0u8; 32]);
+
+    let mut secret_keys = Vec::new();
+    let mut public_keys = Vec::new();
+    let mut authenticators = Vec::new();
+
+    for _ in 0..num_approvers {
+        let sec_key = SecretKey::with_rng(&mut rng);
+        let pub_key = sec_key.public_key();
+
+        secret_keys.push(sec_key);
+        public_keys.push(pub_key);
+    }
+
+    // Create authenticators only for the signers we'll actually use
+    for i in 0..threshold {
+        let authenticator = BasicAuthenticator::<ChaCha20Rng>::new_with_rng(
+            &[(public_keys[i].into(), AuthSecretKey::RpoFalcon512(secret_keys[i].clone()))],
+            rng.clone(),
+        );
+        authenticators.push(authenticator);
+    }
+
+    Ok((secret_keys, public_keys, authenticators))
+}
+
+/// Creates a multisig account with the specified configuration
+fn create_multisig_account(
+    threshold: u32,
+    public_keys: &[PublicKey],
+    asset_amount: u64,
+) -> anyhow::Result<Account> {
+    let approvers: Vec<_> = public_keys.iter().map(|pk| (*pk).into()).collect();
+
+    let multisig_account = AccountBuilder::new([0; 32])
+        .with_auth_component(Auth::Multisig { threshold, approvers })
+        .with_component(BasicWallet)
+        .account_type(AccountType::RegularAccountUpdatableCode)
+        .storage_mode(AccountStorageMode::Public)
+        .with_assets(vec![FungibleAsset::mock(asset_amount)])
+        .build_existing()?;
+
+    Ok(multisig_account)
+}
+
+// ================================================================================================
+// TESTS
+// ================================================================================================
+
+#[tokio::test]
+async fn test_multisig() -> anyhow::Result<()> {
+    // ROLES
+    // - 2 Approvers (multisig signers)
+    // - 1 Multisig Contract
+
+    // Setup keys and authenticators
+    let (_secret_keys, public_keys, authenticators) = setup_keys_and_authenticators(2, 2)?;
+
+    // Create multisig account
+    let multisig_starting_balance = 10u64;
+    let mut multisig_account = create_multisig_account(2, &public_keys, multisig_starting_balance)?;
+
+    let mut mock_chain = MockChainBuilder::with_accounts([multisig_account.clone()])
+        .unwrap()
+        .build()
+        .unwrap();
+
+    // Create input note for the transaction
+    let input_note_asset = FungibleAsset::mock(5);
+    let input_note = create_p2id_note(
+        ACCOUNT_ID_REGULAR_PUBLIC_ACCOUNT_UPDATABLE_CODE.try_into().unwrap(),
+        multisig_account.id(),
+        vec![input_note_asset],
+        NoteType::Public,
+        Default::default(),
+        &mut RpoRandomCoin::new(Word::from([3u32; 4])),
+    )?;
+
+    mock_chain.add_pending_note(OutputNote::Full(input_note.clone()));
+    mock_chain.prove_next_block()?;
```

Comment:
Nit: It should be more concise to use `MockChainBuilder::add_p2id_note` and to avoid having to `prove_next_block`.